src/server/api/routers/match.ts

```diff
@@ -38,13 +38,20 @@ export const matchRouter = createTRPCRouter({
     .subscription(({ ctx, input }) => {
       verifyPlayer(ctx.session.user.id, input.roomId);
 
+      const match = matches.get(input.roomId);
+      const player = match?.players.find(
+        (user) => user.userId === ctx.session.user.id,
+      );
+      if (player) player.isOnline = true;
+
       return observable<ChatMessagePayload>((emit) => {
         const handleEvent = (payload: ChatMessagePayload) => {
           emit.next(payload);
         };
 
         ee.on(matchEvent(input.roomId), handleEvent);
         return () => {
+          if (player) player.isOnline = false;
```

Comment:
Is this correct?
If we have a player then the player is not online?

Reply:
as I understood from Marius the return gets called when the user closes the page, so that's our cue to set it to false until they connect again at which point like 46 sets it back to `true`

Reply:
Yeah, this should be correct, as carlos explained

Reply:
Actually I've just remembered that you don't have to do all the fetching from the map yourself. Just get the player from the return of `verifyPlayer`